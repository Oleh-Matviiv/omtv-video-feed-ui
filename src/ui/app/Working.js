import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const Div = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
`;

const Working = ({loading}) => (
  <Div>{
    loading ? 'loading videos...' : 'failed to load videos'
  }</Div>
);
Working.propTypes = {
  loading: PropTypes.bool.isRequired,
};

export default Working;
