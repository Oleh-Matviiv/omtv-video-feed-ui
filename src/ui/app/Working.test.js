import React from 'react';
import { mount, shallow } from 'enzyme';
import 'jest-styled-components';

import Working, { Div } from './Working';

describe('snapshots', () => {
  it('Working - loading', () => {
    const wrapper = shallow(
      <Working loading={true} />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Working - possibly error has happened', () => {
    const wrapper = shallow(
      <Working loading={false} />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Div', () => {
    const wrapper = shallow(
      <Div>children</Div>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
