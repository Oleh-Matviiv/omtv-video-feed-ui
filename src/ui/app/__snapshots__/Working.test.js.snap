// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Div 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  width: 100%;
  display: flex;
  justify-content: center;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  children
</StyledComponent>
`;

exports[`snapshots Working - loading 1`] = `
<styled.div>
  loading videos...
</styled.div>
`;

exports[`snapshots Working - possibly error has happened 1`] = `
<styled.div>
  failed to load videos
</styled.div>
`;
